module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -1758,6 +1805,33 @@ public function cancelHolds($cancelDetails)
         return $cancelResult;
     }
 
+    /**
+     * Check if request is valid
+     *
+     * This is responsible for determining if an item is requestable
+     *
+     * @param string $id     The record id
+     * @param array  $data   An array of item data
+     * @param array  $patron An array of patron data
+     *
+     * @return array Two entries: 'valid' (boolean) plus 'status' (message to display to user)
+     */
+    public function checkRequestIsValid($id, $data, $patron)
```

Comment:
Implementing this is not strictly necessary; since the rest of the logic should already hide the link, preventing the popup from appearing and this check from being needed.  But it seems reasonable to check here as well, where we can provide an actual error message if it occurs.  Or if someone changes the template.

Also, I imagine there will be other uses for this function soon, i.e. https://openlibraryfoundation.atlassian.net/browse/VUFIND-1698

Reply:
Note that there's an open JIRA ticket about the need to add this method: [VUFIND-1389](https://openlibraryfoundation.atlassian.net/browse/VUFIND-1389). The approach here is different (and perhaps more practical/feasible) than what is proposed there. Do you think we should resolve the ticket when merging this PR? Or perhaps edit it to revise its title/scope?

Reply:
I think VUFIND-1389 is a broader request, and the idea there is correct: to check the circulation rules API, both loan policy (as it says) and request policy.  This PR is a sort of ancillary check for one additional circumstance that would make the request invalid, but definitely doesn't replace VUFIND-1389.  I just got there first.